scripts/gen_syscalls.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import re
+import argparse
+import os
+
+
+api_regex = re.compile(r'''
+__(syscall|syscall_inline)\s+   # __syscall or __syscall_inline
+static\s+inline\s+              # All prototypes are static inline functions
+([^(]+)                         # type and name of system call (split later)
+[(]                             # Function opening parenthesis
+([^)]*)                         # Arg list (split later)
+[)]                             # Closing parenthesis
+''', re.MULTILINE | re.VERBOSE)
+
+typename_regex = re.compile(r'(.*?)([A-Za-z0-9_]+)$')
```

Comment:
Are we expecting to pass arrays as parameters to syscalls? This can't break down things like `int array[]`, `int array[16]`, or `int array[static 16]`. If not, this should be detected and a proper error message should be given; otherwise the script will crash.

Reply:
hadn't considered this. i'll update the script so that this works properly.